### PR TITLE
Add Teams model

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Add `Teams`, `Members` and other teams related entities to the model schema.
* This was a intermediary PR, check the squashed merge commits for the full change log.

### CODE REVIEW

**Comment by vytis** [comment]
Should the #264 merged before this one?

**Comment by daehn** [comment]
@vytis yes

**Comment by daehn** [comment]
@vytis #264 has been merged.

File: Source/Model/User/ZMUser.m [hunk 1]
```diff
@@ -255,7 +258,26 @@ - (NSString *)initials
 
 - (ZMConversation *)oneToOneConversation
 {
-    return self.connection.conversation;
+    // !!!: This is a placeholder as removing this still breaks a lot at the moment.
+
+    // TODO:
```
Comment (daehn) [inline Source/Model/User/ZMUser.m]: We still need to discuss how to handle this, but this can be done separately IMHO.

**Pull request merged**